### ISSUE TITLE
Feedback?

### ISSUE THREAD

**Issue by no-chris** [gregjopa/chord-charts#1]
Hi Greg!

Thanks for checking out on ChordMark! 🙏🏻 
I see that you managed to build a basic app and write a few songs, that's great 😁 

I'm curious if you had any feedback about the format, the software, or anything? 

Thanks a lot!

**Comment by gregjopa** [comment]
Hey @no-chris!

First off, the ChordMark libraries are awesome! Really well done 👏 I haven't seen any other libraries like it. I'm pleased with it and thankful that you open sourced it ❤️ 

I play guitar in my church worship band and thought it would be cool to [build a web app ](https://www.gregjopa.com/chord-charts/index.html) for chord charts using ChordMark to replace the printed out pdf charts we currently use.

I do have some feature requests 😄 

**Support ES Modules** - It would be cool to use unpkg.com with JS modules like so:

```js
import { parseSong, renderSong } from "https://unpkg.com/chord-mark@0.11.0/lib/chord-mark.js";
```

I tried the above syntax but I get an error and ended up referencing chord mark from the global window object. I'm hoping to avoid using a build system and would like to use unpkg.com instead. It would be great to be able to import and apply the themes in a similar manner. Currently I had to copy/paste the generated styles from your demo to get the POC up and running.


**Optional bar lines** - the bar line concept is great and I like how it ties into the different views (no lyrics, lyrics). But there are some use cases where things don't like up nicely. Is it possible to disable/hide the bar lines?

For example, in the song [Glorious Day](https://open.spotify.com/track/3QhL4eWBVv64IycG3i9XTb), the first line in the verse technically has 4 bars of the D chord. And the second line of the verse with the "Who could" lyrics part comes in on beat 4. I wrote it like this which was very similar to the pdf score I was copying from. The only difference is this ChordMark version has the bar lines.
<img width="328" alt="Screen Shot 2022-01-15 at 2 21 59 PM" src="https://user-images.githubusercontent.com/534034/149636930-402725aa-dec1-47fa-b3c0-bc6ec00b252f.png">

For this use case, the bar lines that I coded up are incorrect but the chord placement is nice and readable with the lyrics. Wondering how to best solve for this. My initial thought is hiding the bar lines would be easiest.

That's all the feedback I have right now. I plan on doing some more work with ChordMark and want to scribe more songs and add a feature for switching keys. Thanks for reaching out!

**Comment by no-chris** [comment]
Hello!

Thanks a lot for the feedback and the kind words 🙌🏻 

> For this use case, the bar lines that I coded up are incorrect but the chord placement is nice and readable with the lyrics.  Wondering how to best solve for this.

The correct way to write this song in ChordMark would be the following (based on https://hrbcrichmond.org/wp-content/uploads/2018/08/Glorious-Day-Piano-Vocal-Score.pdf)

<img width="827" alt="Screenshot 2022-01-17 at 15 42 54" src="https://user-images.githubusercontent.com/11467480/149790235-0f026d77-1718-401b-8689-772dfca35c62.png">

The idea is to _always_ encode the rythm information, even if a chord does not change over multiple bars. While this seems (small) extra works, it presents the following advantages:
- it allows more precision on the lyrics placement
- it allows someone who doesn't know the melody to play along with the singers
- it allows to use an alternate rendering for a band member that would not care about the lyrics

<img width="282" alt="Screenshot 2022-01-17 at 15 46 10" src="https://user-images.githubusercontent.com/11467480/149790598-9da2feca-2822-4bbd-abf2-699e20b294aa.png">

That said, you could set the [renderer option](https://github.com/no-chris/chord-mark/blob/master/packages/chord-mark/src/renderer/components/renderSong.js#L40) `printBarSeparators` to `never` or `grids` (= only for lines without lyrics, like the intro) but in that case, I wonder why you would use ChordMark over the classic "chords over lyrics" chord chart format (like Ultimate Guitar)?

> Support ES Modules - It would be cool to use unpkg.com with JS modules like so:

Will look into that! No commitment yet, though ;)

**Comment by no-chris** [comment]
> the "Who could" lyrics part comes in on beat 4

I agree this falls into some kind of void with the current rendering. If that really bother you, you could also write it as:

<img width="839" alt="Screenshot 2022-01-17 at 18 10 42" src="https://user-images.githubusercontent.com/11467480/149812382-cfa93b19-419c-40d6-97f4-206d11c2aac7.png">

**Comment by gregjopa** [comment]
> The idea is to always encode the rythm information, even if a chord does not change over multiple bars.

This is a great point. It's worth it to take the extra time to get the timing right. I plan to keep the barlines.

I really appreciate the advice on how to transcribe Glorious Day. I didn't realize that you can add an underscore in between words. That's really cool! Here's the format I went with:

https://github.com/gregjopa/chord-charts/commit/e9258c05db3710ad0551e78efab5fd8535a5da5e

**Comment by no-chris** [comment]
Great!

I have some more suggestions 😁 

If you want to scribe more songs, you might enjoy some productivity tips 😉 

**Tip 1**: you don't need to repeat chords if they are the same from one section to the other! Just use `_` to position the chords that have been previously defined where they should be (positions are random in my example):

<img width="918" alt="Screenshot 2022-01-18 at 21 30 14" src="https://user-images.githubusercontent.com/11467480/150013931-b242f2ae-9421-4aa0-816f-366b71b6bd2d.png">

**Tip 2**: you don't need to repeat lines if they are the same! Use `%` on a blank line to repeat the last defined chord line, `%%` to repeat the before-last chord line:

<img width="1080" alt="Screenshot 2022-01-18 at 21 33 43" src="https://user-images.githubusercontent.com/11467480/150014252-cc00e757-f412-4743-b8e5-cebb62072b9f.png">

Hope you find those useful!

All those tips (and more) are described in the user guide 😉  
- https://chordmark.netlify.app/docs/reference/chords 
- https://chordmark.netlify.app/docs/reference/sections
